Flaky OpenJDK crashes (Gradle & Bazel)

There's no clear reproduction of this issue, but we're occasionally seeing SIGSEGV failures when running tests:
- https://github.com/oppia/oppia-android/runs/2031235243?check_suite_focus=true
- https://github.com/oppia/oppia-android/pull/2830/checks?check_run_id=2037294748 from #2830 

& others. While the above are Gradle-specific, I believe we've seen this with Bazel as well. Here's one failure result from Gradle:

<details>

```
2021-03-04T13:31:07.6689399Z #
2021-03-04T13:31:07.6690431Z # A fatal error has been detected by the Java Runtime Environment:
2021-03-04T13:31:07.6691221Z #
2021-03-04T13:31:07.6692228Z #  SIGSEGV (0xb) at pc=0x00007efe2d2a4261, pid=9982, tid=0x00007efdffaa2700
2021-03-04T13:31:07.6692888Z #
2021-03-04T13:31:07.6706440Z # JRE version: OpenJDK Runtime Environment (8.0_282-b08) (build 1.8.0_282-b08)
2021-03-04T13:31:07.6707614Z # Java VM: OpenJDK 64-Bit Server VM (25.282-b08 mixed mode linux-amd64 compressed oops)
2021-03-04T13:31:07.6708383Z # Problematic frame:
2021-03-04T13:31:07.6709280Z # V  [libjvm.so+0x92a261]  Node::uncast() const+0x1
2021-03-04T13:31:07.6709697Z #
2021-03-04T13:31:07.6710636Z # Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
2021-03-04T13:31:07.6711317Z #
2021-03-04T13:31:07.6711743Z # An error report file with more information is saved as:
2021-03-04T13:31:07.6712727Z # /home/runner/work/oppia-android/oppia-android/app/hs_err_pid9982.log[thread 139628459652864 also had an error]
2021-03-04T13:31:07.6713236Z 
2021-03-04T13:33:07.6689128Z # [ timer expired, abort... ]
2021-03-04T13:36:00.1688295Z 
2021-03-04T13:36:00.1690914Z org.oppia.android.app.player.state.StateFragmentLocalTest > testStateFragment_nextState_viewFourHints_submitWrongAnswer_noNewHintIsAvailable SKIPPED
2021-03-04T13:36:00.1692819Z 
2021-03-04T13:36:00.1693577Z > Task :app:testDebugUnitTest FAILED
2021-03-04T13:36:00.1695257Z 
2021-03-04T13:36:00.1695531Z 
2021-03-04T13:36:00.1695962Z FAILURE: Build failed with an exception.
2021-03-04T13:36:00.1696838Z Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
2021-03-04T13:36:00.1698555Z Use '--warning-mode all' to show the individual deprecation warnings.
2021-03-04T13:36:00.1699632Z See https://docs.gradle.org/6.8/userguide/command_line_interface.html#sec:command_line_warnings
2021-03-04T13:36:00.1700703Z 
2021-03-04T13:36:00.1701104Z * What went wrong:
2021-03-04T13:36:00.1701914Z Execution failed for task ':app:testDebugUnitTest'.
2021-03-04T13:36:00.1702879Z > Process 'Gradle Test Executor 80' finished with non-zero exit value 134
2021-03-04T13:36:00.1703810Z   This problem might be caused by incorrect test process configuration.
2021-03-04T13:36:00.1704960Z   Please refer to the test execution section in the User Manual at https://docs.gradle.org/6.8/userguide/java_testing.html#sec:test_execution
2021-03-04T13:36:00.1705674Z 
2021-03-04T13:36:00.1705936Z * Try:
2021-03-04T13:36:00.1706674Z Run with --info or --debug option to get more log output. Run with --scan to get full insights.
2021-03-04T13:36:00.1707120Z 
2021-03-04T13:36:00.1707418Z * Exception is:
2021-03-04T13:36:00.1708545Z org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':app:testDebugUnitTest'.
2021-03-04T13:36:00.1710899Z 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$3(ExecuteActionsTaskExecuter.java:186)
2021-03-04T13:36:00.1713004Z 	at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:268)
2021-03-04T13:36:00.1715532Z 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:184)
2021-03-04T13:36:00.1719141Z 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:173)
2021-03-04T13:36:00.1722300Z 	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:109)
2021-03-04T13:36:00.1725638Z 	at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
2021-03-04T13:36:00.1729207Z 	at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:62)
2021-03-04T13:36:00.1733372Z 	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
2021-03-04T13:36:00.1736411Z 	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
2021-03-04T13:36:00.1739325Z 	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
2021-03-04T13:36:00.1742845Z 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
2021-03-04T13:36:00.1746030Z 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
2021-03-04T13:36:00.1748919Z 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
2021-03-04T13:36:00.1758030Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:200)
2021-03-04T13:36:00.1761548Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:195)
2021-03-04T13:36:00.1764652Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:75)
2021-03-04T13:36:00.1767477Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:68)
2021-03-04T13:36:00.1770568Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:153)
2021-03-04T13:36:00.1773741Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:68)
2021-03-04T13:36:00.1776778Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:62)
2021-03-04T13:36:00.1779689Z 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.lambda$call$2(DefaultBuildOperationExecutor.java:76)
2021-03-04T13:36:00.1786372Z 	at org.gradle.internal.operations.UnmanagedBuildOperationWrapper.callWithUnmanagedSupport(UnmanagedBuildOperationWrapper.java:54)
2021-03-04T13:36:00.1790019Z 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:76)
2021-03-04T13:36:00.1793062Z 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
2021-03-04T13:36:00.1795681Z 	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:41)
2021-03-04T13:36:00.1798246Z 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:411)
2021-03-04T13:36:00.1799992Z 114 actionable tasks: 114 executed
2021-03-04T13:36:00.1806916Z 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:398)
2021-03-04T13:36:00.1810083Z 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:391)
2021-03-04T13:36:00.1813373Z 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:377)
2021-03-04T13:36:00.1816019Z 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
2021-03-04T13:36:00.1817997Z 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
2021-03-04T13:36:00.1820310Z 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
2021-03-04T13:36:00.1822650Z 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
2021-03-04T13:36:00.1824641Z 	at org.gradle.execution.plan.DefaultPlanExecutor.process(DefaultPlanExecutor.java:72)
2021-03-04T13:36:00.1827224Z 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph.executeWithServices(DefaultTaskExecutionGraph.java:196)
2021-03-04T13:36:00.1830135Z 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph.execute(DefaultTaskExecutionGraph.java:189)
2021-03-04T13:36:00.1832565Z 	at org.gradle.execution.SelectedTaskExecutionAction.execute(SelectedTaskExecutionAction.java:39)
2021-03-04T13:36:00.1834725Z 	at org.gradle.execution.DefaultBuildWorkExecutor.execute(DefaultBuildWorkExecutor.java:40)
2021-03-04T13:36:00.1836865Z 	at org.gradle.execution.DefaultBuildWorkExecutor.access$000(DefaultBuildWorkExecutor.java:24)
2021-03-04T13:36:00.1838776Z 	at org.gradle.execution.DefaultBuildWorkExecutor$1.proceed(DefaultBuildWorkExecutor.java:48)
2021-03-04T13:36:00.1840742Z 	at org.gradle.execution.DryRunBuildExecutionAction.execute(DryRunBuildExecutionAction.java:49)
2021-03-04T13:36:00.1842863Z 	at org.gradle.execution.DefaultBuildWorkExecutor.execute(DefaultBuildWorkExecutor.java:40)
2021-03-04T13:36:00.1845190Z 	at org.gradle.execution.DefaultBuildWorkExecutor.execute(DefaultBuildWorkExecutor.java:33)
2021-03-04T13:36:00.1848271Z 	at org.gradle.execution.IncludedBuildLifecycleBuildWorkExecutor.execute(IncludedBuildLifecycleBuildWorkExecutor.java:36)
2021-03-04T13:36:00.1851546Z 	at org.gradle.execution.DeprecateUndefinedBuildWorkExecutor.execute(DeprecateUndefinedBuildWorkExecutor.java:44)
2021-03-04T13:36:00.1854851Z 	at org.gradle.execution.BuildOperationFiringBuildWorkerExecutor$ExecuteTasks.run(BuildOperationFiringBuildWorkerExecutor.java:57)
2021-03-04T13:36:00.1857718Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:29)
2021-03-04T13:36:00.1864602Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:26)
2021-03-04T13:36:00.1867238Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:75)
2021-03-04T13:36:00.1869629Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:68)
2021-03-04T13:36:00.1872206Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:153)
2021-03-04T13:36:00.1874958Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:68)
2021-03-04T13:36:00.1877574Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:56)
2021-03-04T13:36:00.1880278Z 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.lambda$run$1(DefaultBuildOperationExecutor.java:71)
2021-03-04T13:36:00.1884126Z 	at org.gradle.internal.operations.UnmanagedBuildOperationWrapper.runWithUnmanagedSupport(UnmanagedBuildOperationWrapper.java:45)
2021-03-04T13:36:00.1887744Z 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:71)
2021-03-04T13:36:00.1892210Z 	at org.gradle.execution.BuildOperationFiringBuildWorkerExecutor.execute(BuildOperationFiringBuildWorkerExecutor.java:42)
2021-03-04T13:36:00.1894878Z 	at org.gradle.initialization.DefaultGradleLauncher.runWork(DefaultGradleLauncher.java:260)
2021-03-04T13:36:00.1897249Z 	at org.gradle.initialization.DefaultGradleLauncher.doClassicBuildStages(DefaultGradleLauncher.java:172)
2021-03-04T13:36:00.1899754Z 	at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:148)
2021-03-04T13:36:00.1902260Z 	at org.gradle.initialization.DefaultGradleLauncher.executeTasks(DefaultGradleLauncher.java:124)
2021-03-04T13:36:00.1904373Z 	at org.gradle.internal.invocation.GradleBuildController$1.create(GradleBuildController.java:72)
2021-03-04T13:36:00.1907574Z 	at org.gradle.internal.invocation.GradleBuildController$1.create(GradleBuildController.java:67)
2021-03-04T13:36:00.1910119Z 	at org.gradle.internal.work.DefaultWorkerLeaseService.withLocks(DefaultWorkerLeaseService.java:213)
2021-03-04T13:36:00.1912763Z 	at org.gradle.internal.invocation.GradleBuildController.doBuild(GradleBuildController.java:67)
2021-03-04T13:36:00.1915754Z 	at org.gradle.internal.invocation.GradleBuildController.run(GradleBuildController.java:56)
2021-03-04T13:36:00.1918818Z 	at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:31)
2021-03-04T13:36:00.1921551Z 	at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
2021-03-04T13:36:00.1924602Z 	at org.gradle.launcher.exec.BuildOutcomeReportingBuildActionRunner.run(BuildOutcomeReportingBuildActionRunner.java:63)
2021-03-04T13:36:00.1927728Z 	at org.gradle.tooling.internal.provider.ValidatingBuildActionRunner.run(ValidatingBuildActionRunner.java:32)
2021-03-04T13:36:00.1930925Z 	at org.gradle.tooling.internal.provider.FileSystemWatchingBuildActionRunner.run(FileSystemWatchingBuildActionRunner.java:77)
2021-03-04T13:36:00.1934410Z 	at org.gradle.launcher.exec.BuildCompletionNotifyingBuildActionRunner.run(BuildCompletionNotifyingBuildActionRunner.java:41)
2021-03-04T13:36:00.1937544Z 	at org.gradle.launcher.exec.RunAsBuildOperationBuildActionRunner$3.call(RunAsBuildOperationBuildActionRunner.java:49)
2021-03-04T13:36:00.1940494Z 	at org.gradle.launcher.exec.RunAsBuildOperationBuildActionRunner$3.call(RunAsBuildOperationBuildActionRunner.java:44)
2021-03-04T13:36:00.1943437Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:200)
2021-03-04T13:36:00.1946923Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:195)
2021-03-04T13:36:00.1950007Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:75)
2021-03-04T13:36:00.1952535Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:68)
2021-03-04T13:36:00.1955105Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:153)
2021-03-04T13:36:00.1957872Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:68)
2021-03-04T13:36:00.1960556Z 	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:62)
2021-03-04T13:36:00.1963282Z 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.lambda$call$2(DefaultBuildOperationExecutor.java:76)
2021-03-04T13:36:00.1966806Z 	at org.gradle.internal.operations.UnmanagedBuildOperationWrapper.callWithUnmanagedSupport(UnmanagedBuildOperationWrapper.java:54)
2021-03-04T13:36:00.1970215Z 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:76)
2021-03-04T13:36:00.1973099Z 	at org.gradle.launcher.exec.RunAsBuildOperationBuildActionRunner.run(RunAsBuildOperationBuildActionRunner.java:44)
2021-03-04T13:36:00.1975853Z 	at org.gradle.launcher.exec.InProcessBuildActionExecuter.lambda$execute$0(InProcessBuildActionExecuter.java:54)
2021-03-04T13:36:00.1978122Z 	at org.gradle.composite.internal.DefaultRootBuildState.run(DefaultRootBuildState.java:86)
2021-03-04T13:36:00.1980677Z 	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:53)
2021-03-04T13:36:00.1983373Z 	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:29)
2021-03-04T13:36:00.1986663Z 	at org.gradle.launcher.exec.BuildTreeScopeLifecycleBuildActionExecuter.lambda$execute$0(BuildTreeScopeLifecycleBuildActionExecuter.java:33)
2021-03-04T13:36:00.1989357Z 	at org.gradle.internal.buildtree.BuildTreeState.run(BuildTreeState.java:49)
2021-03-04T13:36:00.1992059Z 	at org.gradle.launcher.exec.BuildTreeScopeLifecycleBuildActionExecuter.execute(BuildTreeScopeLifecycleBuildActionExecuter.java:32)
2021-03-04T13:36:00.1995907Z 	at org.gradle.launcher.exec.BuildTreeScopeLifecycleBuildActionExecuter.execute(BuildTreeScopeLifecycleBuildActionExecuter.java:27)
2021-03-04T13:36:00.2000090Z 	at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:104)
2021-03-04T13:36:00.2003682Z 	at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:55)
2021-03-04T13:36:00.2006978Z 	at org.gradle.tooling.internal.provider.SubscribableBuildActionExecuter.execute(SubscribableBuildActionExecuter.java:64)
2021-03-04T13:36:00.2010376Z 	at org.gradle.tooling.internal.provider.SubscribableBuildActionExecuter.execute(SubscribableBuildActionExecuter.java:37)
2021-03-04T13:36:00.2014350Z 	at org.gradle.tooling.internal.provider.SessionScopeLifecycleBuildActionExecuter.lambda$execute$0(SessionScopeLifecycleBuildActionExecuter.java:54)
2021-03-04T13:36:00.2017678Z 	at org.gradle.internal.session.BuildSessionState.run(BuildSessionState.java:67)
2021-03-04T13:36:00.2021314Z 	at org.gradle.tooling.internal.provider.SessionScopeLifecycleBuildActionExecuter.execute(SessionScopeLifecycleBuildActionExecuter.java:50)
2021-03-04T13:36:00.2025659Z 	at org.gradle.tooling.internal.provider.SessionScopeLifecycleBuildActionExecuter.execute(SessionScopeLifecycleBuildActionExecuter.java:36)
2021-03-04T13:36:00.2029550Z 	at org.gradle.tooling.internal.provider.GradleThreadBuildActionExecuter.execute(GradleThreadBuildActionExecuter.java:36)
2021-03-04T13:36:00.2032961Z 	at org.gradle.tooling.internal.provider.GradleThreadBuildActionExecuter.execute(GradleThreadBuildActionExecuter.java:25)
2021-03-04T13:36:00.2036576Z 	at org.gradle.tooling.internal.provider.StartParamsValidatingActionExecuter.execute(StartParamsValidatingActionExecuter.java:59)
2021-03-04T13:36:00.2040801Z 	at org.gradle.tooling.internal.provider.StartParamsValidatingActionExecuter.execute(StartParamsValidatingActionExecuter.java:31)
2021-03-04T13:36:00.2045073Z 	at org.gradle.tooling.internal.provider.SessionFailureReportingActionExecuter.execute(SessionFailureReportingActionExecuter.java:55)
2021-03-04T13:36:00.2049639Z 	at org.gradle.tooling.internal.provider.SessionFailureReportingActionExecuter.execute(SessionFailureReportingActionExecuter.java:41)
2021-03-04T13:36:00.2053114Z 	at org.gradle.tooling.internal.provider.SetupLoggingActionExecuter.execute(SetupLoggingActionExecuter.java:47)
2021-03-04T13:36:00.2056087Z 	at org.gradle.tooling.internal.provider.SetupLoggingActionExecuter.execute(SetupLoggingActionExecuter.java:31)
2021-03-04T13:36:00.2058551Z 	at org.gradle.launcher.daemon.server.exec.ExecuteBuild.doBuild(ExecuteBuild.java:65)
2021-03-04T13:36:00.2060790Z 	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:37)
2021-03-04T13:36:00.2063954Z 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
2021-03-04T13:36:00.2067026Z 	at org.gradle.launcher.daemon.server.exec.WatchForDisconnection.execute(WatchForDisconnection.java:39)
2021-03-04T13:36:00.2069851Z 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
2021-03-04T13:36:00.2072731Z 	at org.gradle.launcher.daemon.server.exec.ResetDeprecationLogger.execute(ResetDeprecationLogger.java:29)
2021-03-04T13:36:00.2075593Z 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
2021-03-04T13:36:00.2078731Z 	at org.gradle.launcher.daemon.server.exec.RequestStopIfSingleUsedDaemon.execute(RequestStopIfSingleUsedDaemon.java:35)
2021-03-04T13:36:00.2081722Z 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
2021-03-04T13:36:00.2084086Z 	at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.create(ForwardClientInput.java:78)
2021-03-04T13:36:00.2086147Z 	at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.create(ForwardClientInput.java:75)
2021-03-04T13:36:00.2087591Z 	at org.gradle.util.Swapper.swap(Swapper.java:38)
2021-03-04T13:36:00.2089197Z 	at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:75)
2021-03-04T13:36:00.2091714Z 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
2021-03-04T13:36:00.2094205Z 	at org.gradle.launcher.daemon.server.exec.LogAndCheckHealth.execute(LogAndCheckHealth.java:55)
2021-03-04T13:36:00.2096677Z 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
2021-03-04T13:36:00.2098945Z 	at org.gradle.launcher.daemon.server.exec.LogToClient.doBuild(LogToClient.java:63)
2021-03-04T13:36:00.2101125Z 	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:37)
2021-03-04T13:36:00.2103959Z 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
2021-03-04T13:36:00.2106976Z 	at org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment.doBuild(EstablishBuildEnvironment.java:84)
2021-03-04T13:36:00.2109694Z 	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:37)
2021-03-04T13:36:00.2112148Z 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
2021-03-04T13:36:00.2114831Z 	at org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy$1.run(StartBuildOrRespondWithBusy.java:52)
2021-03-04T13:36:00.2117251Z 	at org.gradle.launcher.daemon.server.DaemonStateCoordinator$1.run(DaemonStateCoordinator.java:297)
2021-03-04T13:36:00.2119282Z 	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
2021-03-04T13:36:00.2121170Z 	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
2021-03-04T13:36:00.2123131Z 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
2021-03-04T13:36:00.2134902Z 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
2021-03-04T13:36:00.2136632Z 	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
2021-03-04T13:36:00.2137928Z 	at java.lang.Thread.run(Thread.java:748)
2021-03-04T13:36:00.2139639Z Caused by: org.gradle.process.internal.ExecException: Process 'Gradle Test Executor 80' finished with non-zero exit value 134
2021-03-04T13:36:00.2141120Z This problem might be caused by incorrect test process configuration.
2021-03-04T13:36:00.2142231Z Please refer to the test execution section in the User Manual at https://docs.gradle.org/6.8/userguide/java_testing.html#sec:test_execution
2021-03-04T13:36:00.2144852Z 	at org.gradle.api.internal.tasks.testing.worker.ForkingTestClassProcessor.stop(ForkingTestClassProcessor.java:182)
2021-03-04T13:36:00.2149245Z 	at org.gradle.api.internal.tasks.testing.processors.RestartEveryNTestClassProcessor.endBatch(RestartEveryNTestClassProcessor.java:77)
2021-03-04T13:36:00.2154086Z 	at org.gradle.api.internal.tasks.testing.processors.RestartEveryNTestClassProcessor.processTestClass(RestartEveryNTestClassProcessor.java:55)
2021-03-04T13:36:00.2157724Z 	at sun.reflect.GeneratedMethodAccessor273.invoke(Unknown Source)
2021-03-04T13:36:00.2159669Z 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2021-03-04T13:36:00.2161358Z 	at java.lang.reflect.Method.invoke(Method.java:498)
2021-03-04T13:36:00.2162757Z 	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
2021-03-04T13:36:00.2165167Z 	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
2021-03-04T13:36:00.2169643Z 	at org.gradle.internal.dispatch.FailureHandlingDispatch.dispatch(FailureHandlingDispatch.java:30)
2021-03-04T13:36:00.2172469Z 	at org.gradle.internal.dispatch.AsyncDispatch.dispatchMessages(AsyncDispatch.java:87)
2021-03-04T13:36:00.2174742Z 	at org.gradle.internal.dispatch.AsyncDispatch.access$000(AsyncDispatch.java:36)
2021-03-04T13:36:00.2176568Z 	at org.gradle.internal.dispatch.AsyncDispatch$1.run(AsyncDispatch.java:71)
2021-03-04T13:36:00.2178548Z 	at org.gradle.internal.concurrent.InterruptibleRunnable.run(InterruptibleRunnable.java:42)
2021-03-04T13:36:00.2181993Z 	at org.gradle.internal.operations.CurrentBuildOperationPreservingRunnable.run(CurrentBuildOperationPreservingRunnable.java:42)
2021-03-04T13:36:00.2196293Z 	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
2021-03-04T13:36:00.2198521Z 	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
2021-03-04T13:36:00.2200641Z 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
2021-03-04T13:36:00.2202473Z 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
2021-03-04T13:36:00.2204505Z 	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
2021-03-04T13:36:00.2206053Z 	at java.lang.Thread.run(Thread.java:748)
```

</details>

Note that this seems related to https://bugs.openjdk.java.net/browse/JDK-8183200, and https://github.com/AdoptOpenJDK/openjdk-support/issues/55 has some interesting information. It seems to correlate to large amounts of forking, and if one fork fails the whole thing fails. That might explain why we're seeing this more now since we enabled test forking in Gradle a few months back.

Interestingly, the logs suggest that we're using JDK 1.8 even though we're setting up 1.9 in our configuration: https://github.com/oppia/oppia-android/blob/develop/.github/workflows/main.yml#L90. Need to figure out what's going on with that.

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -35,20 +35,27 @@ jobs:
        with:
           java-version: 1.9
       
-     - name: Check Java version
-       run: java -version
+     - name: Check Java version & path
+       run: |
+         which java
+         java -version
+         echo "Java home: $JAVA_HOME"
+         $JAVA_HOME/bin/java -version
 
      - name: Install Dependencies
        if: steps.cache.outputs.cache-hit != 'true'
-       run: ./gradlew --full-stacktrace androidDependencies
+       run: ./gradlew --full-stacktrace androidDependencies -Dorg.gradle.java.home=$JAVA_HOME
+
+     - name: Check Gradle environment
+       run: ./gradlew -Dorg.gradle.java.home=$JAVA_HOME -v
 
      - name: Build App
        # We require 'sudo' to avoid an error of the existing android sdk. See https://github.com/actions/starter-workflows/issues/58
-       run: sudo ./gradlew --full-stacktrace assembleDebug
+       run: sudo ./gradlew --full-stacktrace assembleDebug -Dorg.gradle.java.home=$JAVA_HOME
 
      - name: Utility tests
        # We require 'sudo' to avoid an error of the existing android sdk. See https://github.com/actions/starter-workflows/issues/58
-       run:  sudo ./gradlew --full-stacktrace :utility:testDebugUnitTest
+       run:  sudo ./gradlew --full-stacktrace :utility:testDebugUnitTest -Dorg.gradle.java.home=$JAVA_HOME
      - name: Upload Utility Test Reports
        uses: actions/upload-artifact@v2
        if: ${{ always() }} # IMPORTANT: Upload reports regardless of status
@@ -58,7 +65,7 @@ jobs:
 
      - name: Domain tests
        # We require 'sudo' to avoid an error of the existing android sdk. See https://github.com/actions/starter-workflows/issues/58
-       run:  sudo ./gradlew --full-stacktrace :domain:testDebugUnitTest
+       run:  sudo ./gradlew --full-stacktrace :domain:testDebugUnitTest -Dorg.gradle.java.home=$JAVA_HOME
      - name: Upload Domain Test Reports
        uses: actions/upload-artifact@v2
        if: ${{ always() }} # IMPORTANT: Upload reports regardless of status
@@ -68,7 +75,7 @@ jobs:
 
      - name: Testing tests
        # We require 'sudo' to avoid an error of the existing android sdk. See https://github.com/actions/starter-workflows/issues/58
-       run:  sudo ./gradlew --full-stacktrace :testing:testDebugUnitTest
+       run:  sudo ./gradlew --full-stacktrace :testing:testDebugUnitTest -Dorg.gradle.java.home=$JAVA_HOME
      - name: Upload Testing Test Reports
        uses: actions/upload-artifact@v2
        if: ${{ always() }} # IMPORTANT: Upload reports regardless of status
@@ -94,14 +101,21 @@ jobs:
         uses: actions/setup-java@v1
         with:
           java-version: 1.9
-      
-      - name: Check Java version
-        run: java -version
+
+      - name: Check Java version & path
+        run: |
+          which java
+          java -version
+          echo "Java home: $JAVA_HOME"
+          $JAVA_HOME/bin/java -version
+
+      - name: Check Gradle environment
+        run: ./gradlew -Dorg.gradle.java.home=$JAVA_HOME -v
 
       - name: Robolectric tests - App Module
         # We require 'sudo' to avoid an error of the existing android sdk. See https://github.com/actions/starter-workflows/issues/58
         run: |
-          sudo ./gradlew --full-stacktrace :app:testDebugUnitTest
+          sudo ./gradlew --full-stacktrace :app:testDebugUnitTest -Dorg.gradle.java.home=$JAVA_HOME
       - name: Upload App Test Reports
         uses: actions/upload-artifact@v2
         if: ${{ always() }} # IMPORTANT: Upload reports regardless of status

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -34,6 +34,9 @@ jobs:
        uses: actions/setup-java@v1
        with:
           java-version: 1.9
+      
+     - name: Check Java version
+       run: java -version
 
      - name: Install Dependencies
        if: steps.cache.outputs.cache-hit != 'true'
@@ -91,6 +94,9 @@ jobs:
         uses: actions/setup-java@v1
         with:
           java-version: 1.9
+      
+      - name: Check Java version
+        run: java -version
 
       - name: Robolectric tests - App Module
         # We require 'sudo' to avoid an error of the existing android sdk. See https://github.com/actions/starter-workflows/issues/58

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -128,7 +128,11 @@ dependencies {
       'org.jetbrains.kotlinx:kotlinx-coroutines-android:1.2.1',
       'org.mockito:mockito-core:2.7.22',
   )
-  compileOnly 'javax.annotation:javax.annotation-api:1.3.2'
+  compileOnly(
+      'jakarta.xml.bind:jakarta.xml.bind-api:2.3.2',
+      'javax.annotation:javax.annotation-api:1.3.2',
+      'org.glassfish.jaxb:jaxb-runtime:2.3.2',
+  )
   testImplementation(
       'androidx.test:core:1.2.0',
       'androidx.test.espresso:espresso-contrib:3.1.0',

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -128,6 +128,7 @@ dependencies {
       'org.jetbrains.kotlinx:kotlinx-coroutines-android:1.2.1',
       'org.mockito:mockito-core:2.7.22',
   )
+  compileOnly 'javax.annotation:javax.annotation-api:1.3.2'
   testImplementation(
       'androidx.test:core:1.2.0',
       'androidx.test.espresso:espresso-contrib:3.1.0',

--- a/build.gradle
+++ b/build.gradle
@@ -29,22 +29,3 @@ allprojects {
 task clean(type: Delete) {
   delete rootProject.buildDir
 }
-
-subprojects{
-  afterEvaluate {
-    // kapt doesn't propagate source version properly so even if we set JDK to be 9, if we
-    // have a local JDK version of 9+, we see a dagger bug in generating annotations
-    // that only occurs in JDK 9+, this is the workaround:
-    // (Lots of useful information here: https://github.com/google/dagger/issues/1449)
-    if (project.hasProperty('kapt')) {
-      // Reference for 'kapt' DSL: https://kotlinlang.org/docs/reference/kapt.html#java-compiler-options
-      kapt {
-        // we expect this closure to run over a org.jetbrains.kotlin.gradle.plugin.KaptExtension
-        javacOptions {
-          option("-source", "8")
-          option("-target", "8")
-        }
-      }
-    }
-  }
-}

--- a/data/build.gradle
+++ b/data/build.gradle
@@ -70,7 +70,11 @@ dependencies {
       'com.squareup.moshi:moshi-kotlin:1.11.0',
       'org.jetbrains.kotlinx:kotlinx-coroutines-android:1.2.2',
   )
-  compileOnly 'javax.annotation:javax.annotation-api:1.3.2'
+  compileOnly(
+      'jakarta.xml.bind:jakarta.xml.bind-api:2.3.2',
+      'javax.annotation:javax.annotation-api:1.3.2',
+      'org.glassfish.jaxb:jaxb-runtime:2.3.2',
+  )
   testImplementation(
       'android.arch.core:core-testing:1.1.1',
       'androidx.test.ext:junit:1.1.1',

--- a/data/build.gradle
+++ b/data/build.gradle
@@ -70,6 +70,7 @@ dependencies {
       'com.squareup.moshi:moshi-kotlin:1.11.0',
       'org.jetbrains.kotlinx:kotlinx-coroutines-android:1.2.2',
   )
+  compileOnly 'javax.annotation:javax.annotation-api:1.3.2'
   testImplementation(
       'android.arch.core:core-testing:1.1.1',
       'androidx.test.ext:junit:1.1.1',

--- a/domain/build.gradle
+++ b/domain/build.gradle
@@ -70,6 +70,7 @@ dependencies {
       'com.google.firebase:firebase-crashlytics:17.0.0',
       "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
   )
+  compileOnly 'javax.annotation:javax.annotation-api:1.3.2'
   testImplementation(
       'android.arch.core:core-testing:1.1.1',
       'androidx.test.espresso:espresso-core:3.2.0',

--- a/domain/build.gradle
+++ b/domain/build.gradle
@@ -70,7 +70,11 @@ dependencies {
       'com.google.firebase:firebase-crashlytics:17.0.0',
       "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
   )
-  compileOnly 'javax.annotation:javax.annotation-api:1.3.2'
+  compileOnly(
+      'jakarta.xml.bind:jakarta.xml.bind-api:2.3.2',
+      'javax.annotation:javax.annotation-api:1.3.2',
+      'org.glassfish.jaxb:jaxb-runtime:2.3.2',
+  )
   testImplementation(
       'android.arch.core:core-testing:1.1.1',
       'androidx.test.espresso:espresso-core:3.2.0',

--- a/testing/build.gradle
+++ b/testing/build.gradle
@@ -63,6 +63,7 @@ dependencies {
       project(":domain"),
       project(":utility"),
   )
+  compileOnly 'javax.annotation:javax.annotation-api:1.3.2'
   testImplementation(
       'androidx.lifecycle:lifecycle-livedata-ktx:2.2.0-alpha03',
       'androidx.test.ext:junit:1.1.1',

--- a/testing/build.gradle
+++ b/testing/build.gradle
@@ -63,7 +63,11 @@ dependencies {
       project(":domain"),
       project(":utility"),
   )
-  compileOnly 'javax.annotation:javax.annotation-api:1.3.2'
+  compileOnly(
+      'jakarta.xml.bind:jakarta.xml.bind-api:2.3.2',
+      'javax.annotation:javax.annotation-api:1.3.2',
+      'org.glassfish.jaxb:jaxb-runtime:2.3.2',
+  )
   testImplementation(
       'androidx.lifecycle:lifecycle-livedata-ktx:2.2.0-alpha03',
       'androidx.test.ext:junit:1.1.1',

--- a/utility/build.gradle
+++ b/utility/build.gradle
@@ -72,7 +72,11 @@ dependencies {
       'com.google.firebase:firebase-crashlytics:17.0.0',
       "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version",
   )
-  compileOnly 'javax.annotation:javax.annotation-api:1.3.2'
+  compileOnly(
+      'jakarta.xml.bind:jakarta.xml.bind-api:2.3.2',
+      'javax.annotation:javax.annotation-api:1.3.2',
+      'org.glassfish.jaxb:jaxb-runtime:2.3.2',
+  )
   testImplementation(
       'androidx.test.ext:junit:1.1.1',
       'com.google.dagger:dagger:2.24',

--- a/utility/build.gradle
+++ b/utility/build.gradle
@@ -72,6 +72,7 @@ dependencies {
       'com.google.firebase:firebase-crashlytics:17.0.0',
       "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version",
   )
+  compileOnly 'javax.annotation:javax.annotation-api:1.3.2'
   testImplementation(
       'androidx.test.ext:junit:1.1.1',
       'com.google.dagger:dagger:2.24',

